src/deroff.rs

```diff
@@ -366,70 +367,70 @@ impl Deroffer {
         })
     }
 
-    fn comment<'a>(&self, mut s: &'a str) -> &'a str {
-        while Self::str_at(&s, 0) != "\n" {
-            s = self.skip_char(&s, 1);
-        }
-        s
-    }
-
-    fn skip_char<'a>(&self, s: &'a str, amount: usize) -> &'a str {
-        s.get(amount..).unwrap_or("")
+    fn skip_char(&mut self, amount: usize) {
+        self.s.drain(..amount);
     }
 
-    fn skip_leading_whitespace<'a>(&self, s: &'a str) -> &'a str {
-        s.trim_start()
+    fn skip_leading_whitespace(&mut self) {
+        self.s = self.s.trim_start().to_owned();
     }
 
-    fn str_at(string: &str, idx: usize) -> &str {
-        // Note: If we don't care about strings with multi-byte chars, the
-        // following would suffice:
-        // s.get(idx..idx + 1).unwrap_or("")
-        //
-        // Note: We're not yet sure whether our roff inputs will generally be
-        // ASCII or UTF-8. If they are ASCII (and can be treated as containing
-        // only single-byte characters), it would be faster to just use `get()`
-        string
-            .char_indices()
+    fn str_at(&self, idx: usize) -> &str {
+        let s = &self.s;
+        s.char_indices()
             .skip(idx)
             .next()
-            .map(|(idx, charr)| &string[idx..(idx + charr.len_utf8())]) // Okay to directly index based on idx/charr construction.
-            .unwrap_or_default()
+            .map(|(i, c)| &s[i..(i + c.len_utf8())])
+            .unwrap_or("")
     }
 
-    fn is_white<'a>(s: &'a str, idx: usize) -> bool {
-        match Self::str_at(s, idx) {
+    fn is_white(&self, idx: usize) -> bool {
+        match self.str_at(idx) {
             "" => false,
             c => c.chars().all(|c| c.is_whitespace()),
         }
     }
 
-    fn digit(s: &str, idx: usize) -> bool {
-        match Self::str_at(s, idx) {
+    // AKA `prch`
```

Comment:
Maybe use `///` since it will be generated in the documentation?
```suggestion
    /// AKA `prch`
```